doc/modules/impute.rst

```diff
@@ -313,3 +313,12 @@ wrap this in a :class:`Pipeline` with a classifier (e.g., a
   >>> results = clf.predict(X_test)
   >>> results.shape
   (100,)
+
+Estimators that handle NaN values
+=================================
+
+Some estimators are designed to handle NaN values without preprocessing. 
+Below is the list of these estimators, classified by type 
+(cluster, regressor, classifier, transform) :
+
+.. allow_nan::
```

Comment:
Instead of having a single directive call, it would be possible to have something like the following:

```rst
.. list_estimators::
    :filter_type: classifier
    :estimator_tag: allow_nan

.. list_estimators::
    :filter_type: regressor
    :estimator_tag: allow_nan

.. list_estimators::
    :filter_type: cluster
    :estimator_tag: allow_nan

.. list_estimators::
    :filter_type: transformer
    :estimator_tag: allow_nan
```

In which case it would be better to rename the `doc/sphinxext/allow_nan.py ` file to `doc/sphinxext/list_estimators.py ` and the `AllowNan` directive to `ListEstimators`.